armory/scenarios/video_update.py

```diff
@@ -0,0 +1,197 @@
+"""
+Classifier evaluation within ARMORY
+
+Scenario Contributor: MITRE Corporation
+"""
+
+import logging
+from typing import Optional
+
+import numpy as np
+from tqdm import tqdm
+
+from armory.utils.config_loading import (
+    load_dataset,
+    load_model,
+    load_attack,
+    load_adversarial_dataset,
+    load_label_targeter,
+)
+from armory.utils import metrics
+from armory.scenarios.base import Scenario
+
+logger = logging.getLogger(__name__)
+
+
+class Context:
+    def __init__(self):
+        self.nb_classes = 101
+        self.classes = tuple(range(self.nb_classes))
+        self.x_dimensions = (None, None, 240, 320, 3)
+        self.default_float = np.float32
+        self.quantization = 255
+
+
+context = Context()
+
+
+def check_input(batch):
+    if batch.dtype != np.uint8:
+        raise ValueError(f"input batch dtype {batch.dtype} != np.uint8")
+    elif batch.ndim != len(context.x_dimensions):
+        raise ValueError(f"input batch dim {batch.ndim} != {len(context.x_dimensions)}")
+    for dim, (source, target) in enumerate(zip(batch.shape, context.x_dimensions)):
+        pass
+
+
+def dataset_canonical_preprocessing(batch):
+    assert batch.dtype == np.uint8
+    assert batch.ndim == 5
+    assert batch.shape[2:] == context.x_dimensions[2:]
+
+    batch = batch.astype(context.default_float) / context.quantization  # 255
+    assert batch.dtype == context.default_float
+    assert batch.max() <= 1.0
+    assert batch.min() >= 0.0
+
+    return batch
+
+
+class Ucf101(Scenario):
+    def _evaluate(
```

Comment:
Can we decouple some loading logic as standalone callable methods? It would make us easier to tweak/visualize a scenario in Jupyter.

FYI, we have an internal implementation that has
```
_load_defenses()
_load_attacks()
_load_test_data()
_load_classifier_with_defense()
```

Reply:
Can you explain in more detail (or provide an example) of how these differ from the imported methods:
```
from armory.utils.config_loading import (
    load_dataset,
    load_model,
    load_attack,
    load_adversarial_dataset,
    load_label_targeter,
)
```

Reply:
That's a great idea! I'll include it as a separate issue and PR, though.

Reply:
Added https://github.com/twosixlabs/armory/issues/741